core/controlplane/cluster/cluster.go

```diff
@@ -157,7 +157,7 @@ func (c *Cluster) Assets() (cfnstack.Assets, error) {
 
 	return cfnstack.NewAssetsBuilder(c.StackName(), c.StackConfig.S3URI).
 		Add("userdata-controller", c.UserDataController).
-		Add("userdata-worker", c.UserDataWorker).
```

Comment:
No!
Although it is required only by a node pool stack, we had been unnecessarily uploading `user-worker` here for a control-plane stack, too.